src/parsers/manifest/dash/parse_mpd.ts

```diff
@@ -274,21 +274,29 @@ function parseCompleteIntermediateRepresentation(
     minimumTime = contentStart !== undefined            ? contentStart :
                   parsedPeriods[0]?.start !== undefined ? parsedPeriods[0].start :
                                                           0;
-    let maximumTime : number | undefined;
-    if (contentEnd !== undefined) {
-      maximumTime = contentEnd;
-    } else if (mediaPresentationDuration !== undefined) {
+    let maximumTime = Infinity;
+    if (mediaPresentationDuration !== undefined &&
+        mediaPresentationDuration < maximumTime) {
       maximumTime = mediaPresentationDuration;
-    } else if (parsedPeriods[parsedPeriods.length - 1] !== undefined) {
+    }
+    if (parsedPeriods[parsedPeriods.length - 1] !== undefined) {
       const lastPeriod = parsedPeriods[parsedPeriods.length - 1];
-      maximumTime = lastPeriod.end ??
-                    (lastPeriod.duration !== undefined ?
-                      lastPeriod.start + lastPeriod.duration :
-                      undefined);
+      const lastPeriodEnd = lastPeriod.end ??
```

Comment:
What is the difference between the `lastPeriodEnd` and the `contentEnd` ?

Reply:
I answered you elsewhere but will there also:
  - `lastPeriodEnd` is the maximum position defined for a given Period (e.g. a DASH Period)
  - `contentEnd` is the ending position of the last segment in that Period

Those can be different in subtle conditions, though there is works to simplify everything (#942)